### PR TITLE
Add Circuit latest

### PR DESCRIPTION
**Pull request by ignic**
After making all changes to the cask:

- [x] `brew cask audit --download {{cask_file}}` is error-free.
- [x] `brew cask style --fix {{cask_file}}` reports no offenses.
- [x] The commit message includes the cask’s name and version.
- [x] The submission is for [a stable version](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#finding-a-home-for-your-cask) or [documented exception](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#but-there-is-no-stable-version).

Additionally, if **adding a new cask**:

- [x] Named the cask according to the [token reference].
- [x] `brew cask install {{cask_file}}` worked successfully.
- [x] `brew cask uninstall {{cask_file}}` worked successfully.
- [x] Checked there are no [open pull requests] for the same cask.
- [x] Checked the cask was not [already refused].
- [x] Checked the cask is submitted to [the correct repo].

[token reference]: https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/token_reference.md
[open pull requests]: https://github.com/Homebrew/homebrew-cask/pulls
[already refused]: https://github.com/Homebrew/homebrew-cask/search?q=is%3Aclosed&type=Issues
[the correct repo]: https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#finding-a-home-for-your-cask

### CODE REVIEW

**Comment by vitorgalvao** [comment]
@core-code This one has an `app-update.yml`:

```
provider: generic
url: ''
updaterCacheDirName: circuit-updater
```

Are you able to figure out the correct endpoint?

**Comment by ignic** [comment]
> @core-code This one has an `app-update.yml`:
> 
> ```
> provider: generic
> url: ''
> updaterCacheDirName: circuit-updater
> ```
> 
> Are you able to figure out the correct endpoint?

@vitorgalvao I'm afraid I am not able to figure out this endpoint.
The web app only shows the link (that is in the url stanza) to download the Mac client, but no additional information about updates or versions.

**Comment by ignic** [comment]
I have added the `auto_updates` stanza as this seems to be possible within the app.

**Comment by core-code** [comment]
i've tried to sniff the appcast url but it isn't possible without invoking the update check. and this doesn't seem possible without logging in. i've made an account with them in order to log in but it just says 'not available now try later' or something.

File: Casks/circuit.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+cask 'circuit' do
+  version :latest
+  sha256 :no_check
+
+  # eu.yourcircuit.com was verified as official when first introduced to the cask
```
Comment (vitorgalvao) [inline Casks/circuit.rb]: I’m not finding the app on the website (I though I found it before). Where is it?

Comment (core-code) [reply Casks/circuit.rb]: perhaps it is related to the 'Circuit is unavailable' message when trying to log-in

Comment (ignic) [reply Casks/circuit.rb]: Once logged in the web app, the link to download the Mac client app is in Profile > Settings > Extensions

circuit

Comment (vitorgalvao) [reply Casks/circuit.rb]: > Once logged in the web app

That makes it a [walled build](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#finding-a-home-for-your-cask), I’m afraid, so [we cannot accept it](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#unofficial-vendorless-and-walled-builds).

Thank you anyway for the submission.

**Comment by aderuelle** [comment]
Hi ! You might try those URLs :

https://eu.yourcircuit.com/desktop/available/mac/*

`{"available":true,"version":"1.2.8202"}`

https://eu.yourcircuit.com/desktop/available/mac/1028202

`{"available":false,"version":"1.2.8202"}`

https://eu.yourcircuit.com/desktop/updates/mac/1028202

`{"url":"https://eu.yourcircuit.com/desktop/releases/mac/Circuit-1.2.8202-mac.zip","version":"1.2.8202"}`

Hope this helps

**Comment by vitorgalvao** [comment]
> Hi ! You might try those URLs :

None of those solve the issue of this being a walled build; they’re all from the same host as the problematic URL in the PR.

**Comment by aderuelle** [comment]
Understood.

@ignic could submit its cask to [homebrew-unofficial](https://github.com/alehouse/homebrew-unofficial) though, as suggested [here](https://github.com/Homebrew/homebrew-cask/blob/master/doc/faq/rejected_casks.md).

**Pull request closed without merging**